### PR TITLE
Update example to bootstrap-loader

### PR DESCRIPTION
**Pull request by alex35mil**
@shakacode/team-react_on_rails 

Just pushed rewritten [`bootstrap-loader`](https://github.com/shakacode/bootstrap-loader/tree/alex/bootstrap-4) to npm. Bootstrap 3 & 4 supported. Feel free to test this out.

screen shot 2015-11-14 at 00 54 29

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/bootstrap-sass-loader-example/6)

### CODE REVIEW

**Comment by alexfedoseev** [comment]
There is no React here yet. Just simple html to say hello world with new loader. Will update it to React + CSS Modules this weekend.

**Comment by rstudner** [comment]
Does it make sense at all to have both react/CDs module examples and just the worlds simplest - you only want bootstrap example?

**Comment by alexfedoseev** [comment]
@rstudner The main point is that I can't show example of CSS Modules + Bootstrap without React.

**Comment by rstudner** [comment]
hah!  Very good point :)

**Comment by justin808** [comment]
@alexfedoseev I will create a new repo for you call bootstrap-loader-css-modules-example.

**Comment by justin808** [comment]
https://github.com/shakacode/bootstrap-loader-css-modules-example created.

File: app/bootstrap/customizations.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Customize Bootstrap variables
+
+$btn-primary-bg: #29d;
+$btn-primary-color: #fff;
+$btn-primary-border: $btn-primary-bg;
```
Comment (justin808) [inline app/bootstrap/customizations.scss]: Why not put all these in the pre-customizations file?

File: .bootstraprc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+# This param can also be set in webpack config:
+#   entry: 'bootstrap?extractStyles!./app'
+extractStyles: false
+# env:
+#   development:
+#     extractStyles: false
+#   production:
+#     extractStyles: true
+
+# preBootstrapCustomizations: ./app/bootstrap/pre-customizations.scss
```
Comment (justin808) [inline .bootstraprc]: ?

File: README.md [hunk 3]
```diff
@@ -1,41 +1,15 @@
...
-2. [webpack.config.js](https://github.com/justin808/bootstrap-sass-loader-example/blob/master/webpack.config.js)
-
-This example is based on this example for bootstrap, less version: [bootstrap-webpack-example](http://bline.github.io/bootstrap-webpack-example/).
-
-See `gulpfile.js` for more information on gulp tasks to run, such as `gulp test`, `gulp webpack`, and `gulp build`.
-
-
-Testing Changes in the Bootstrap Sass Loader
-=======================================================
-1. See this article [Debugging NodeJs and Webpack Loaders](http://forum.railsonmaui.com/t/debugging-nodejs-and-webpack-loaders/142)
```
Comment (justin808) [inline README.md]: We should make sure these testing tips are somewhere accessible. I rely on these notes.

File: app/markup/bootstrap-prod.html [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html>
+
+  <head>
+    <title>Bootstrap Production Server</title>
+    <link rel="stylesheet" href="/assets/app.css">
```
Comment (justin808) [inline app/markup/bootstrap-prod.html]: We might want a comment to call this out.

File: webpack.dev.config.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoErrorsPlugin(),
+  ],
+
+  module: {
+    loaders: [
+      { test: /\.css$/, loaders: [ 'style', 'css', 'postcss' ] },
+      { test: /\.scss$/, loaders: [ 'style', 'css', 'postcss', 'sass' ] },
+      { test: /\.(woff2?|ttf|eot|svg)$/, loader: 'url?limit=10000' },
+
```
Comment (justin808) [inline webpack.dev.config.js]: // Use one of these:

File: webpack.prod.config.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const path = require('path');
+const autoprefixer = require('autoprefixer');
+
+module.exports = {
+
+  entry: 'bootstrap?extractStyles!./app/components/app',
```
Comment (justin808) [inline webpack.prod.config.js]: This could probably use a comment.

File: package.json [hunk 7]
```diff
@@ -1,51 +1,16 @@
...
-    "url-loader": "^0.5.6",
-    "webpack": "^1.10.1",
-    "webpack-dev-server": "^1.10.1"
-  },
   "scripts": {
-    "test": "gulp test"
+    "start": "nodemon --watch app/markup --ext html server.dev.js",
+    "clean": "rm -rf public/",
+    "build": "npm run clean && webpack --config webpack.prod.config.js",
+    "prod": "npm run build && nodemon --watch app/markup --ext html server.prod.js"
```
Comment (justin808) [inline package.json]: Can you explain this commands or give me a doc reference?

**Comment by justin808** [comment]
Hot reload did not seem to be working.

**Comment by alexfedoseev** [comment]
@justin808 CSS is hot reloadable. If you're about JS, I need some JS code to try this out, maybe it needs additional setup (like React does). Anyway it's not first priority here, will take a look later.

screen recording 2015-11-14 at 05 50 pm

File: app/bootstrap/customizations.scss (hunk 1 above)
Comment (alexfedoseev) [inline app/bootstrap/customizations.scss]: I was just playing around, testing things. Moved to pre-customizations file.

File: .bootstraprc (hunk 2 above)
Comment (alexfedoseev) [inline .bootstraprc]: Uncommented.

File: README.md (hunk 3 above)
Comment (alexfedoseev) [inline README.md]: Yep, I just removed it for now to highlight how to run this setup -> add relevant parts from it later.

File: app/markup/bootstrap-prod.html (hunk 4 above)
Comment (alexfedoseev) [inline app/markup/bootstrap-prod.html]: :+1:

File: webpack.dev.config.js (hunk 5 above)
Comment (alexfedoseev) [inline webpack.dev.config.js]: :+1:

File: webpack.prod.config.js (hunk 6 above)
Comment (alexfedoseev) [inline webpack.prod.config.js]: :+1:

File: package.json (hunk 7 above)
Comment (alexfedoseev) [inline package.json]: It just reloads node server `server.xxx.js` when `html` file in `app/markup` has been modified. [Link](https://github.com/remy/nodemon).

**Comment by alexfedoseev** [comment]
Updated.

**Comment by alexfedoseev** [comment]
This branch updated + posted CSS Modules example: https://github.com/shakacode/bootstrap-loader-css-modules-example

**Comment by alex35mil** [comment]
I don't remember what is it.

**Pull request closed without merging**